userdiff.c

```diff
@@ -174,6 +174,8 @@ PATTERNS("bibtex", "(@[a-zA-Z]{1,}[ \t]*\\{{0,1}[ \t]*[^ \t\"@',\\#}{~%]*).*$",
 	 "[={}\"]|[^={}\" \t]+"),
 PATTERNS("tex", "^(\\\\((sub)*section|chapter|part)\\*{0,1}\\{.*)$",
 	 "\\\\[a-zA-Z@]+|\\\\.|[a-zA-Z0-9\x80-\xff]+"),
+PATTERNS("ts", "^[\t ]*(constructor|public|private|protected).*$",
+	 "^[\t ]*(const|let|var)[^()]*=[\t ]*\\([^()]*\\)[\t ]*=>*$"),
```

Comment:
Seems like you intended `>.*` because otherwise this is saying: "accept any number of `>` characters"

If you accept this suggestion it will say: "accept rest of line", which will pull in the EOL-context for the hunk-header.
```suggestion
	 "^[\t ]*(const|let|var)[^()]*=[\t ]*\\([^()]*\\)[\t ]*=>.*$"),
```